test/functionalTest/cases/1048_subscription_cache/cache_population_at_init.test

```diff
@@ -0,0 +1,243 @@
+        "isPattern": "true",
+        "type": "T1"
+    },
+    {
+        "id":   "E3",
+        "type": "T1"
+    }    
+  ],
+  "attributes": [
+    "A1",
+    "A2"
+  ],
+  "reference": "http://localhost:'${LISTENER_PORT}'/notify",
+  "duration": "P1M",
+  "throttling": "PT1S",
+  "notifyConditions": [
+    {
+        "type": "ONCHANGE",
+        "condValues": [
+            "A1",
+            "A2"
+        ]
+    }
+  ]
+}'
+orionCurl --url /v1/subscribeContext --payload "$payload" --json
+echo
+echo
+
+
+echo "03. Subscribe to E3.*,E4,E5.*/A1+A2+A3"
+echo "======================================"
+payload='{
+  "entities": [
+    {
+        "id":   "E3.*",
+        "isPattern": "true",
+        "type": "T1"
+    },
+    {
+        "id":   "E4",
+        "type": "T1"
+    },
+    {
+        "id":   "E5.*",
+        "isPattern": "true",
+        "type": "T1"
+    }    
+  ],
+  "attributes": [
+    "A1",
+    "A2",
+    "A3"
+  ],
+  "reference": "http://localhost:'${LISTENER_PORT}'/notify",
+  "duration": "P1M",
+  "throttling": "PT1S",
+  "notifyConditions": [
+    {
+        "type": "ONCHANGE",
+        "condValues": [
+            "A1",
+            "A2",
+            "A3"
+        ]
+    }
+  ]
+}'
+orionCurl --url /v1/subscribeContext --payload "$payload" --json
+echo
+echo
+
+
+echo "04. Restart broker and wait a second"
+echo "===================================="
+brokerStop -v CB
+echo "Broker Stopped"
+sleep 1
+brokerStart CB > /dev/null
+echo "Broker Restarted"
+sleep 1
+echo
+echo
+
+
+echo "05. Statistics"
+echo "=============="
+orionCurl --url /statistics --json
+echo
+echo
+
+
+--REGEXPECT--
+01. Subscribe to E1.*/A1
+========================
+HTTP/1.1 200 OK
+Content-Length: 135
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "subscribeResponse": {
+        "duration": "P1M", 
+        "subscriptionId": "REGEX([0-9a-f]{24})", 
+        "throttling": "PT1S"
+    }
+}
+
+
+02. Subscribe to E2.*,E3/A1+A2
+==============================
+HTTP/1.1 200 OK
+Content-Length: 135
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "subscribeResponse": {
+        "duration": "P1M", 
+        "subscriptionId": "REGEX([0-9a-f]{24})", 
+        "throttling": "PT1S"
+    }
+}
+
+
+03. Subscribe to E3.*,E4,E5.*/A1+A2+A3
+======================================
+HTTP/1.1 200 OK
+Content-Length: 135
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "subscribeResponse": {
+        "duration": "P1M", 
+        "subscriptionId": "REGEX([0-9a-f]{24})", 
+        "throttling": "PT1S"
+    }
+}
+
+
+04. Restart broker and wait a second
+====================================
+Broker Stopped
+Broker Restarted
+
+
+05. Statistics
+==============
+HTTP/1.1 200 OK
+Content-Length: 176
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "orion": {
+        "jsonRequests": "1", 
+        "measuring_interval_in_secs": "REGEX(\d)", 
+        "statisticsRequests": "1", 
+        "subCacheEntries": "3", 
```

Comment:
Good idea to inlude counters and statistics about cache usage! Very useful for debugging.

NTC